### DISCUSSION TITLE
No module named 'elastalert_modules'

### DISCUSSION THREAD

**Discussion by pravyn** [jertel/elastalert2#1275] [Q&A] [answered]
Hi,

We have running prod Elastalert2, now we have copied new rule with match_enhancement. But we are getting below error

![image](https://github.com/jertel/elastalert2/assets/19923046/b5736b61-3e5c-4a3a-8a21-dfc96fc81e3b)

`Sep 21 01:12:12 server01 python3[14309]: ERROR:elastalert:Could not load rule /opt/elastalert2/rules/systems/disks/os_disk_alert_less_5gb.yaml: Could not import module elastalert_modules.drop_if_not_exists_in_list.DropIfNotExistsInList: No module named 'elastalert_modules'
`

But when i run the rule manually, it working perfectly. Looks like while running elastalert as service, module error is coming up. Can you please help on this.

Thanks,
Praveen

**Comment by jertel** [comment]
Where did you place the custom enhancement file?

**Reply by pravyn** [reply]
@jertel  My custom module location /opt/elastalert2/elastalert_modules/drop_if_not_exists_in_list.py

**Reply by jertel** [reply]
How did you install ElastAlert 2?

**Reply by pravyn** [reply]
We used git clone method.

**Reply by jertel** [reply]
Did you follow this procedure? https://elastalert2.readthedocs.io/en/latest/running_elastalert.html#as-a-python-package

**Reply by pravyn** [reply]
Yes @jertel  we have followed the above wiki.

**Reply by jertel** [reply]
Then most likely you need to place that new module file in the appropriate /usr/lib/python3 subtree.

**Reply by pravyn** [reply]
I have copied the file as shown below but no luck. When i check the elastalert status, the related rule is not picking up.

<img width="988" alt="image" src="https://github.com/jertel/elastalert2/assets/19923046/c40033ed-a6ae-4143-8292-e7074fcf7d0c">

**Reply by jertel** [reply, marked as answer]
It doesn't go in the root of that dir. It will go in the _appropriate_ subtree. Look at site-packages, and find elastalert2. Follow the enhancements documentation.

**Reply by pravyn** [reply]
I will do some testing and update you @jertel . Thanks for your help.
